### PR TITLE
New post for additional Java to create multiple VSD objects

### PR DESCRIPTION
**Pull request by paquel**
Added new 'md' file and some image files for a new posting that describes how to construct and remove a VSD network layout of Enterprise, Domain, Zone and Subnet objects.

### CODE REVIEW

File: _posts/2018-02-16-Java-Time2.md [hunk 1]
```diff
@@ -0,0 +1,1564 @@
...
+                enterprise.delete(); // 13
+            } catch (RestException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+}
+{% endhighlight %}
+
+Note that in the constructor at Line 1, there is a `VSDSession` argument.  A `VSDSsssion` object establishes a secure HTTP connection to a VSD instance.  It is similar to how you would create a connection to a database before running SQL queries.  Before calling methods in this class, we will assume that a VSD session has already been established and started by `NetworkBuilder`.
```
Comment (pdellaert) [inline _posts/2018-02-16-Java-Time2.md]: `VSDSsssion` -> `VSDSession`

File: _posts/2018-02-16-Java-Time2.md [hunk 2]
```diff
@@ -0,0 +1,1564 @@
...
+
+In the `fetchEnterprise` method at Line 4, we instantiate a `Me` object to serve as the root for all of our enterprise specific network objects.  Line 5 sets up a search filter based on the name of our enterprise.  In Line 6, we create an instance of `EnterprisesFetcher`.  The VSPK provides "Fetchers" for all entity types in the VSD to offer convenience methods for operating on collections of VSD objects.  Line 7 calls the `getFirst` method of the Fetcher to retrieve the Enterprise object matching the name in the search filter.
+
+Line 8 of the `createEnterprise` method is similar to Line 4 of the `fetchEnterprise` method.  It provides us with a 'Me' root object onto which we can attach a new Enterprise object.  In Line 9, we call the constructor of the VSPK Enterprise class to create a brand new Enterprise object and then in Line 10 we assign our enterprise name.  Finally, in Line 11, we create a new child Enterprise object.
```
Comment (pdellaert) [inline _posts/2018-02-16-Java-Time2.md]: `'Me'` -> `` `Me` ``

File: _posts/2018-02-16-Java-Time2.md [hunk 3]
```diff
@@ -0,0 +1,1564 @@
...
+Here is the updated version of the `NetworkBuilder` class that integrates with `NetworkEnterpriseHandler`.
+
+{% highlight java %}
+import java.util.List;
+import net.nuagenetworks.bambou.RestException;
+import net.nuagenetworks.vspk.v5_0.Enterprise;
+import net.nuagenetworks.vspk.v5_0.VSDSession;
+
+public class NetworkBuilder {
+    private String url = "https://135.228.4.108:8443"; // 1
```
Comment (pdellaert) [inline _posts/2018-02-16-Java-Time2.md]: Don't use a 135.... IP in documentation, use one from https://tools.ietf.org/html/rfc5737

Suggestion: `198.51.100.108`

File: _posts/2018-02-16-Java-Time2.md [hunk 4]
```diff
@@ -0,0 +1,1564 @@
...
+
+{% highlight java %}
+import java.util.List;
+import net.nuagenetworks.bambou.RestException;
+import net.nuagenetworks.vspk.v5_0.DomainTemplate;
+import net.nuagenetworks.vspk.v5_0.Enterprise;
+import net.nuagenetworks.vspk.v5_0.VSDSession;
+
+public class NetworkBuilder {
+    private String url = "https://135.228.4.108:8443";
```
Comment (pdellaert) [inline _posts/2018-02-16-Java-Time2.md]: Don't use a 135.... IP in documentation, use one from https://tools.ietf.org/html/rfc5737

Suggestion: `198.51.100.108`

File: _posts/2018-02-16-Java-Time2.md [hunk 5]
```diff
@@ -0,0 +1,1564 @@
...
+{% highlight java %}
+import java.util.List;
+import net.nuagenetworks.bambou.RestException;
+import net.nuagenetworks.vspk.v5_0.Domain;
+import net.nuagenetworks.vspk.v5_0.DomainTemplate;
+import net.nuagenetworks.vspk.v5_0.Enterprise;
+import net.nuagenetworks.vspk.v5_0.VSDSession;
+
+public class NetworkBuilder {
+    private String url = "https://135.228.4.108:8443";
```
Comment (pdellaert) [inline _posts/2018-02-16-Java-Time2.md]: Don't use a 135.... IP in documentation, use one from https://tools.ietf.org/html/rfc5737

Suggestion: `198.51.100.108`

File: _posts/2018-02-16-Java-Time2.md [hunk 6]
```diff
@@ -0,0 +1,1564 @@
...
+import java.util.List;
+import net.nuagenetworks.bambou.RestException;
+import net.nuagenetworks.vspk.v5_0.Zone;
+import net.nuagenetworks.vspk.v5_0.Domain;
+import net.nuagenetworks.vspk.v5_0.DomainTemplate;
+import net.nuagenetworks.vspk.v5_0.Enterprise;
+import net.nuagenetworks.vspk.v5_0.VSDSession;
+
+public class NetworkBuilder {
+    private String url = "https://135.228.4.108:8443";
```
Comment (pdellaert) [inline _posts/2018-02-16-Java-Time2.md]: Don't use a 135.... IP in documentation, use one from https://tools.ietf.org/html/rfc5737

Suggestion: `198.51.100.108`

File: _posts/2018-02-16-Java-Time2.md [hunk 7]
```diff
@@ -0,0 +1,1564 @@
...
+import net.nuagenetworks.bambou.RestException;
+import net.nuagenetworks.vspk.v5_0.Subnet;
+import net.nuagenetworks.vspk.v5_0.Zone;
+import net.nuagenetworks.vspk.v5_0.Domain;
+import net.nuagenetworks.vspk.v5_0.DomainTemplate;
+import net.nuagenetworks.vspk.v5_0.Enterprise;
+import net.nuagenetworks.vspk.v5_0.VSDSession;
+
+public class NetworkBuilder {
+    private String url = "https://135.228.4.108:8443";
```
Comment (pdellaert) [inline _posts/2018-02-16-Java-Time2.md]: Don't use a 135.... IP in documentation, use one from https://tools.ietf.org/html/rfc5737

Suggestion: `198.51.100.108`

File: _posts/2018-02-16-Java-Time2.md [hunk 8]
```diff
@@ -0,0 +1,1564 @@
...
+Network requires Subnet: MyLittleWebSubnet1
+Enterprise MyLittleEnterprise already exists
+Template MyLittleDomainTemplate already exists
+Domain MyLittleDomain already exists
+Zone MyLittleWebTierZone already exists
+Subnet MyLittleWebSubnet1 already exists
+{% endhighlight %}
+
+# Removing VSD Objects
+In order to remove VSD objects, there are dependency rules that must be followed, based on parent-child relationships within the object tree.  For example, to remove a Zone object, there must not be any Subnet child objects attached.  Likewise, to remove a Domain object, you must first remove its Zone objects.  We also want to handle cases where we have more than one child object for a parent.  For simplicity, we will assume for this exercise that our network falls under a single common enterprise.
```
Comment (pdellaert) [inline _posts/2018-02-16-Java-Time2.md:1080]: Just a note that this is not exactly true. You can actually delete a whole domain without first deleting the subnet and zones, as long as there are no vPorts present.

However, for the sake of your blog post, no need to make a change here :)

File: _posts/2018-02-16-Java-Time2.md [hunk 9]
```diff
@@ -0,0 +1,1564 @@
...
+            try {
+                template.delete(); // 13
+            } catch (RestException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+{% endhighlight %}
+
+In Line 9 of the `fetchAllTemplatesForEnterprise` method, we gather all Domain Template children for a given Enterprise into a Fetcher.  Then in Lines 10 and 11, we c
```
Comment (pdellaert) [inline _posts/2018-02-16-Java-Time2.md]: Don't use a 135.... IP in documentation, use one from https://tools.ietf.org/html/rfc5737

Suggestion: `198.51.100.108`

**Review by pdellaert — CHANGES_REQUESTED** [review]
One typo, one small remark and one remark about ip addresses which i've mentioned everywhere it was appropriate (so repeated)

Overal great blog post! Thanks :)

**Comment by paquel** [comment]
Hi Philippe:  Thank you so much for the quick review.

**Comment by paquel** [comment]
@pdellaert Thanks Philippe.  Made suggested changes

**Review by pdellaert — APPROVED** [review]

**Pull request merged**